### PR TITLE
Rework tests for `lat_bins_area()`

### PR DESCRIPTION
**Pull request by etiennebacher**
This PR revamps tests for `lat_bins_area()`. It:

* splits tests into more `test_that()` code blocks
* adds more tests with wrong inputs
* uses snapshot tests to check output

Some of the new snapshots show "bad" error messages, those will be improved once `rlang` is used in the package (#163). 

Part of #161 

## Checklist before requesting a review
- [x] I have read palaeoverse's [standards and structure](https://palaeoverse.palaeoverse.org/articles/structure-and-standards.html)
- [x] I have performed a self-review of my code.
- [ ] I have added function documentation.
- [ ] I have provided sufficient examples of implementation.
- [ ] If it is a core feature, I have added thorough tests.

### CODE REVIEW

File: tests/testthat/_snaps/lat_bins_area.md [hunk 1]
```diff
@@ -1,11 +1,61 @@
...
+    Code
+      lat_bins_area(n = 6)
+    Output
+        bin       min       mid       max         area area_prop
+      1   1  41.81031  65.90516  90.00000 8.501075e+13 0.1666667
+      2   2  19.47122  30.64077  41.81031 8.501075e+13 0.1666667
+      3   3   0.00000   9.73561  19.47122 8.501075e+13 0.1666667
+      4   4 -19.47122  -9.73561   0.00000 8.501075e+13 0.1666667
+      5   5 -41.81031 -30.64077 -19.47122 8.501075e+13 0.1666667
+      6   6 -90.00000 -65.90516 -41.81031 8.501075e+13 0.1666667
```
Comment (etiennebacher) [inline tests/testthat/_snaps/lat_bins_area.md:31]: Using a snapshot is a gold standard here: we can check the number of rows, columns, column names, and ensure that we know if values change for some reason in the future. Worth noting that snapshot tests are slightly slower to run, but I don't think it's a big problem here.

We could have an `expect_equal()` for this but it would just be very annoying to type all the decimals.

File: tests/testthat/_snaps/lat_bins_area.md [hunk 2]
```diff
@@ -1,11 +1,61 @@
...
       Error in `lat_bins_area()`:
       ! `n` should be a numeric.
 
+---
+
+    Code
+      lat_bins_area(n = -1)
+    Condition
+      Error in `data.frame()`:
+      ! arguments imply differing number of rows: 3, 1, 0
```
Comment (etiennebacher) [inline tests/testthat/_snaps/lat_bins_area.md:60]: This (and a few other error messages) will be improved by `rlang` in the refactor.

File: tests/testthat/test-lat_bins_area.R [hunk 3]
```diff
@@ -1,20 +1,44 @@
...
+test_that("different radius affects the results", {
+  orig <- lat_bins_area(n = 6)
+  new <- lat_bins_area(n = 6, r = 6000)
+  expect_all_false(orig$area == new$area)
+})
+
+# TODO: what should happen here? Currently it returns area = 0 and area_prop = NaN
+# test_that("lat_bins_aread handles case where min == max", {
+#   lat_bins_area(min = 90, max = 90)
+# })
```
Comment (etiennebacher) [inline tests/testthat/test-lat_bins_area.R]: I guess this should be an error (i.e. the the `max >= min` condition should actually be `max > min`)? What do you think?

``` r
palaeoverse::lat_bins_area(3, min = 90, max = 90)
#>   bin min mid max area area_prop
#> 1   1  90  90  90    0       NaN
#> 2   2  90  90  90    0       NaN
#> 3   3  90  90  90    0       NaN
```

File: tests/testthat/test-lat_bins_area.R [hunk 4]
```diff
@@ -1,20 +1,44 @@
...
+
   expect_snapshot(lat_bins_area(plot = "TRUE"), error = TRUE)
+  expect_snapshot(lat_bins_area(plot = logical(0)), error = TRUE)
+
   expect_snapshot(lat_bins_area(r = "Earth"), error = TRUE)
-  expect_snapshot(lat_bins_area(n = 3.5), error = TRUE)
+  expect_snapshot(lat_bins_area(r = numeric(0)), error = TRUE)
+
+  # TODO: I suppose this should error? Currently it doesn't
+  # expect_snapshot(lat_bins_area(r = -1), error = TRUE)
```
Comment (etiennebacher) [inline tests/testthat/test-lat_bins_area.R]: Can you confirm this should error?

``` r
palaeoverse::lat_bins_area(3, r = -1)
#>   bin       min           mid       max    area area_prop
#> 1   1  19.47122  5.473561e+01  90.00000 4188790 0.3333333
#> 2   2 -19.47122 -3.552714e-15  19.47122 4188790 0.3333333
#> 3   3 -90.00000 -5.473561e+01 -19.47122 4188790 0.3333333
```

File: tests/testthat/test-lat_bins_area.R (hunk 3 above)
Comment (etiennebacher) [reply tests/testthat/test-lat_bins_area.R]: Note that this errors in `lat_bins_degrees()`:

``` r
palaeoverse::lat_bins_degrees(min = 30, max = 30)
#> Error in `palaeoverse::lat_bins_degrees()`:
#> ! `min` should be less than `max`
```

Comment (LewisAJones) [reply tests/testthat/test-lat_bins_area.R]: Yes, this should be an error, the output isn't exactly useful!

File: tests/testthat/test-lat_bins_area.R (hunk 4 above)
Comment (LewisAJones) [reply tests/testthat/test-lat_bins_area.R]: Yeah, this is nonsensical (`r` = the radius of the Earth in km) but I am not sure if and/or when we should error. I think we should still keep `r` as a 'free' input value. Erroring when less than or equal to 0 probably makes a lo of sense, and perhaps adding a warning if less than/greater than standard values used for the radius?

**Review by LewisAJones — APPROVED** [review]
Good job on picking up some edge cases/bugs. The reworked tests look good to me! We will want to revisit some of this during the refactoring, but for now, this looks great.

File: tests/testthat/test-lat_bins_area.R (hunk 4 above)
Comment (willgearty) [reply tests/testthat/test-lat_bins_area.R]: I think for now we should error when equal to or less than 0. Implementing better warnings sounds like a good small thing for someone to work on in the future.

**Review by willgearty — APPROVED** [review]
Looks good, thanks! (coverage tests failing here as well?)

**Comment by etiennebacher** [comment]
I have checked the test coverage output and also followed the steps detailed in the CI logs (download the Github artifact and use the snapshots contained in it) and can't find any problem. Lewis found another issue when mixing test coverage and snapshots (https://github.com/r-lib/devtools/issues/2698) so I think this is just a false positive.

Thanks both for your review!
